doc/nrf/config_and_build/dfu/index.rst

```diff
@@ -10,20 +10,39 @@ Device Firmware Updates
-You can transfer the updated images to the device in two ways:
+.. note::
+  The choice of bootloader affects how firmware updates can be performed.
+  While specifics regarding bootloader features such as out-of-the-box support for various DFU methods may vary, it's generally recognized that:
 
-* Wired, where updates are sent through a wired connection, like UART, or delivered by connecting a flash device.
-* Over-the-air (OTA), where updates are sent through a wireless connection, like Bluetooth® Low Energy.
+  * MCUboot and its supporting libraries and middleware is designed with flexibility in mind and supports various protocols and methods for firmware updates.
+    However, the extent of supported features, like USB-DFU class, may vary and should be confirmed with the latest MCUboot documentation or source code.
+  * The nRF Secure Immutable Bootloader (NSIB) allows for firmware updates but might not include comprehensive support for all types of firmware updates compared to MCUboot.
+    Specifics of NSIB’s capabilities, particularly in terms of direct DFU support, are subject to change and should be verified with the latest documentation.
 
-The testing and booting process depends on the choice of bootloader.
+  This distinction is crucial for developing your firmware update strategy and selecting the appropriate bootloader for your device's needs.
+
+The testing and booting process depends on the choice of bootloader, but also on the design of the application.
 Generally, bootloaders support two types of updates:
 
 * Direct updates, with an in-place substitution of the image.
-  In this case, it is the bootloader that must transfer the update image.
+  In this case, it is typically the bootloader that must transfer the update image.
 * Background updates, where the updated image is obtained and stored by the application, but the update is completed by the bootloader after the device reboots.
 
+While the bootloader is essential for the final steps of the DFU process, including image verification and swapping, the application often plays a crucial role in receiving and staging updates, especially when using the Simple Management Protocol (SMP).
+For instance, in systems that use MCUboot, the application may be responsible for receiving update packages through SMP and staging them for the bootloader.
+
+These can be explained like so:
+
+* Bootloader capabilities typically refer to the bootloader's ability to receive, verify, and activate either new or already-received firmware images.
+* Application-level update mechanisms involve the application's role in collecting new firmware/candidate images and preparing them for installation by the bootloader.
+
+You can transfer the updated images to the device in two ways:
+
+* Wired, where updates are sent through a wired connection, like UART, or delivered by connecting a flash device.
+* Over-the-air (OTA), where updates are sent through a wireless connection, like Bluetooth® Low Energy.
+
```

Comment:
This is all good, but I think the structure could be improved. I'm getting a bit lost between all the lists.
Would an image be a viable option?

Reply:
I will look into how this can be improved - including perhaps adding an image

Reply:
@greg-fer So far I have added subheadings - let me know if this improves it? Otherwise, please let me know of any other feedback.

I have been discussing initial ideas for an image to show the DFU workflow, but I don't think I will add it in this PR.

Reply:
Sorry for late update, the list of review requests grew a bit and this PR got lost at the bottom.